client/my-sites/site-settings/controller.js

```diff
@@ -75,6 +76,12 @@ module.exports = {
 			return;
 		}
 
+		// redirect seo tab to general for Jetpack sites
+		if ( site.jetpack && endsWith( basePath, 'seo' ) ) {
```

Comment:
A cleaner way to check for the section is `( context.params.section === 'seo' )` and you can get rid of the `endsWith` dependency. When the routes are declared, the placeholders get passed into the called controller methods in the `context` object that page.js creates. So this is populated form the route here https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/site-settings/index.js#L29.